core/exec-operand.c

```diff
@@ -401,3 +411,31 @@ static void lai_write_buffer(lai_nsnode_t *handle, lai_variable_t *source) {
         n += m;
     }
 }
+
+// lai_read_buffer(): Reads from a BufferField.
+static void lai_read_buffer(lai_variable_t *dest, lai_nsnode_t *handle) {
+	lai_nsnode_t *buffer_handle = handle->bf_node;
+
+	size_t offset = handle->bf_offset;
+	size_t size = handle->bf_size;
+	uint8_t *data = lai_exec_buffer_access(&buffer_handle->object);
+	dest->type = LAI_INTEGER;
+	dest->integer = 0;
+
+	int n = 0;
+	while(n < size) {
+		int bit = (offset + n) & 7;
+		int m = size - n;
+		if (m > (8 - bit))
+			m = 8 - bit;
+		LAI_ENSURE(m); // read at least one bit.
+		uint8_t mask = (1 << m) - 1;
+		uint8_t cur_byte = data[(offset + n) >> 3];
+		uint8_t to_write = 0;
+		to_write |= ((cur_byte & mask));
```

Comment:
This declaration is not necessary (at least move the assignment to the same line).